test/cpp/c10d/ProcessGroupNCCLTest.cpp

```diff
@@ -761,6 +761,7 @@ class ProcessGroupNCCLTest : public ::testing::Test {
   int size_{1};
 };
 
+/*
```

Comment:
why are we doing this? The test is indeed needed.

Reply:
I agree. This is just a draft PR trying the in-place constructor. I can copy the updated test in your PR if you'd like this PR to be a formal one. Let me know.